packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js

```diff
@@ -280,68 +218,51 @@ const reducer = (state, action) =>
-         * which is an array. Hence why we replace those fields.
-         */
-        const mergeDataWithPreparedRelations = relationalFieldPaths
-          .map((path) => path.split('.'))
-          .reduce((acc, currentPaths) => {
-            const [componentName] = currentPaths;
-
-            const existingComponents = castArray(acc[componentName] || []);
-            existingComponents.reduce((result, currentEntry) => {
-              if (!isNil(get(currentEntry, [`__temp_key__`]))) {
-                return result;
-              }
-
-              set(currentEntry, [`__temp_key__`], getMaxTempKey(result) + 1);
-
-              return result;
-            }, existingComponents);
-
-            if (
-              state.modifiedData &&
-              get(state.modifiedData, componentName) &&
-              /**
-               * Only replace the state if the entity ids are the same.
-               */
-              state.modifiedData?.id === initialValues?.id
-            ) {
-              /**
-               * this will be null on initial load, however subsequent calls
-               * will have data in them correlating to the names of the relational fields.
-               *
-               * We also merge the fetched data so that things like `id` for components can be copied over
-               * which would be `undefined` in the `browserState`.
-               */
-              const currentState = cloneDeep(get(state.modifiedData, componentName));
-              set(acc, componentName, merge(currentState, get(initialValues, componentName)));
-            } else if (
-              repeatableComponentPaths.includes(componentName) ||
-              dynamicZonePaths.includes(componentName) ||
-              componentPaths.includes(componentName)
-            ) {
-              /**
-               * if the componentName is a repeatable field or dynamic zone we collect the list of paths e.g.
-               * ["repeatable_single_component_relation","categories"] and then reduce this
-               * recursively
-               */
-              const findleaf = findLeafByPathAndReplace(currentPaths.slice(-1)[0], []);
-              currentPaths.reduce(findleaf, acc);
-            } else {
-              set(acc, currentPaths, []);
+        const findAllRelationsAndReplaceWithEmptyArray = findAllAndReplace(
+          components,
+          (value) => value.type === 'relation',
+          (_, { path }) => {
+            if (state.modifiedData?.id === data.id && get(state.modifiedData, path)) {
+              return get(state.modifiedData, path);
             }
 
-            return acc;
-          }, data);
+            return [];
+          }
+        );
+
+        const mergedDataWithPreparedRelations = findAllRelationsAndReplaceWithEmptyArray(
+          data,
+          attributes
+        );
+
+        const findAllDynamicAndRegularComponentsAndReplaceWithTempKey = findAllAndReplace(
```

Comment:
Are dynamic components = repeatable components? And are regular components non-repeatable components?

It is very nit-picky, but if so I'd suggest to rename this to

```suggestion
        const findComponentsAndReplaceWithTempKey = findAllAndReplace(
```